EducationalAssignments/UndoPartOne.md

```diff
@@ -162,24 +161,25 @@ OBJC = "objc"
 
 
 class LPFile():
-    def __init__(self,filename,create):
+    def __init__(self, filename, create):
         # globals
-        mycontext['debug'] = False   
-        self.LPfile = openfile(filename,create)
-        self.pending_data = ""
-        self.pending_offset = 0
+        mycontext['debug'] = False
+        self.LPfile = openfile(filename, create)
+        self.pending_data = None
+        self.pending_offset = None
 
     def readat(self, bytes, offset):
         # Read from the file using the sandbox's readat...
         return self.LPfile.readat(bytes, offset)
 
-    def writeat(self,data,offset):
-        self.LPfile.writeat(data, offset)
-        self.pending_data += data
-        self.pending_offset = offset + len(data)
+    def writeat(self, data, offset):
+        self.LPfile.writeat(self.pending_data, self.pending_offset)
+        self.pending_data = data
+        self.pending_offset = offset
 
     def undo(self):
-        self.LPfile.writeat("", self.pending_offset)
+        self.pending_data = data
```

Comment:
Won't these give an exception?

Reply:
If we are running along with the test case, it throws `RepyArgumentError` but I clarified it here
- https://github.com/SeattleTestbed/docs/pull/83/files#diff-3631efb7544152c4840e37a47ef9c3a3304f532656e22fb006cc6fd7925dfb54R86-R87
- https://github.com/SeattleTestbed/docs/pull/83/files#diff-3631efb7544152c4840e37a47ef9c3a3304f532656e22fb006cc6fd7925dfb54R247

Reply:
I followed this method from the previous year assignment, the idea is the faulty implementation gives an error but fixing it (the correct implementation) would pass the test without any error.

Reply:
Right, but this seems to have a bug because "data" isn't scoped in the function.  It's more of a syntax error than one of algorithmic logic.

Reply:
Prof @JustinCappos, how about 
```py
        self.pending_data = None
        self.pending_offset = None
```

Reply:
I updated the code as per your suggestion.